baseline_testing/testcase_bestest_air_scenario.py

```diff
@@ -0,0 +1,131 @@
+import time
+
+
+def run(modelname, scenario, start_time = 0, warmup_period = 0, length = 48*3600,  plot=False):
+    """Run test case.
+    Parameters
+    ----------
+    modelname: str
+        Name of the testcase.
+        
+    scenario: dict, optional
+        Dictionary defining the predefined test scenario.
+        {'time_period': str, 'electricity_price': str}.
+        If specified, start_time, warmup_period, and length not used.
+        
+    start_time: int, optional
+        Simulation start time in seconds from midnight January 1st.
+        Not used if scenario defined.
+        
+    warmup_period: int, optional
+        Simulation warmup-period in seconds before start_time.
+        Not used if scenario defined.
+        
+    length: int, optional
+        Simulation duration in seconds (e.g., 24*3600 is a 1 day simulation).
+        Not used if scenario defined.
+
+    plot : bool, optional
+        True to plot timeseries results.
+        Default is False.
+
+    Returns
+    -------
+    kpi : dict
+        Dictionary of core KPI names and values.
+        {kpi_name : value}
+    res : dict
+        Dictionary of trajectories of inputs and outputs.
+    custom_kpi_result: dict
+        Dictionary of tracked custom KPI calculations.
+        Empty if no customized KPI calculations defined.
+
+    """
+
+    # RUN THE CONTROL TEST
+    # --------------------
+    control_module = 'examples.python.controllers.{}'.format(modelname)
+    step = 300
+    
+    # RUN THE CONTROL TEST
+    # --------------------
+    #start_time, warmup_period, length will not be used if scenario is specified.
+    kpi, df_res, custom_kpi_result, forecasts = control_test(control_module,start_time,warmup_period,length,
+                                                                  scenario=scenario,step=step)
+
+    # POST-PROCESS RESULTS
+    # --------------------
+    time = df_res.index.values / 3600  # convert s --> hr
+    zone_temperature = df_res['zon_reaTRooAir_y'].values - 273.15  # convert K --> C
+    # Plot results
+    if plot:
+        try:
+            from matplotlib import pyplot as plt
+            import numpy as np
+            plt.figure(1)
+            plt.title('Zone Temperature')
+            plt.plot(time, zone_temperature)
+            plt.plot(time, 20 * np.ones(len(time)), '--')
+            plt.plot(time, 23 * np.ones(len(time)), '--')
+            plt.ylabel('Temperature [C]')
+            plt.xlabel('Time [hr]')
+            plt.show()
+        except ImportError:
+            print("Cannot import numpy or matplotlib for plot generation")
+
+    return kpi, df_res, custom_kpi_result
+
+
+if __name__ == "__main__":
+
+    modelname='bestest_air'
+    
+    #Run baseline control testing of specified scenarios
+    # --------------------
+    price_elec=['constant','dynamic','highly_dynamic']
+    time_period=['peak_heat_day','typical_heat_day', 'peak_cool_day', 'typical_cool_day', 'mix_day']
```

Comment:
I suggest this information come from a config file for the given test case, per comment above.

Reply:
It has been moved into a config file with the following structure:

```
{
    "save_kpi_results": true,
    "save_measurements": true,
    "run_user_defined_test": false,
    "bestest_air": {
        "electricity_price": ["constant", "dynamic", "highly_dynamic"],
        "time_period": ["peak_heat_day", "typical_heat_day", "peak_cool_day", "typical_cool_day", "mix_day"]
    },
    "bestest_hydronic_heat_pump": {
        "electricity_price": ["constant", "dynamic", "highly_dynamic"],
        "time_period": ["peak_heat_day", "typical_heat_day"]
    },
    "multizone_office_simple_air": {
        "electricity_price": ["constant", "dynamic", "highly_dynamic"],
        "time_period": ["peak_heat_day", "typical_heat_day", "peak_cool_day", "typical_cool_day", "mix_day"]
    },
    "multizone_residential_hydronic": {
        "electricity_price": ["constant", "dynamic", "highly_dynamic"],
        "time_period": ["peak_heat_day", "typical_heat_day"]
    },
    "singlezone_commercial_hydronic": {
        "electricity_price": ["constant", "dynamic", "highly_dynamic"],
        "time_period": ["peak_heat_day", "typical_heat_day"]
    },
    "twozone_apartment_hydronic": {
        "electricity_price": ["constant", "dynamic"],
        "time_period": ["peak_heat_day"]
    }
}
```